### PR TITLE
gitignore support

### PR DESCRIPTION
**Pull request by PThorpe92**
https://github.com/schollz/croc/issues/508
Writing my own parser just was too bloated even for the utils, plus I wouldn't want to use something that wasn't production tested yet on a project like this.

I actually wrote some very extensive tests (a few times the size of the contribution) but didn't submit them because they had to create files / edit files with relative paths due to the nature of testing a .gitignore
All the tests passed/worked, but I have been limited in my ability to test the built binary (solo dev with one pc), so please let me know if there's an issue

### CODE REVIEW

**Comment by schollz** [comment]
This looks amazing, nice work. I thought it was ambitious to write your own package! But I applaud that, if you ever feel like it I'd be happy to incorporate a hand rolled lib. But this PR looks nice and clean and well done, will give it a check over this week.

**Comment by PThorpe92** [comment]
Thank you, yeah without handwriting the parsing it turned out to be quite simple. Very clean, and well written codebase btw. Great program all together I look forward to working on it further

**Comment by schollz** [comment]
Not sure this is working - here's how I test:

```
go build -v
mkdir mm && mv croc mm
cd mm
touch .gitignore
echo "croc" > .gitignore
touch test.txt
echo "test" > test.txt
./croc send *
```

the result is


```
Sending 2 files (9.0 MB)         
Code is: zackary
On the other computer run

croc XX
```

where I would have expected it to say "Sending 1 file"

**Comment by PThorpe92** [comment]
strange.. In croc_test I edited and rewrote the "TestCrocLocal" to do just that in a few diff ways and each time it would send only what wasn't in the .gitignore, and without it it would send everything.

Thank you, I'm looking into it now

File: README.md
Comment (stefins) [inline README.md:1]: Please revert this file deletion.

Comment (PThorpe92) [reply README.md:1]: my fault one of the tests generates a readme in a src folder I thought this was that deletion that accidentally got committed

**Comment by schollz** [comment]
still running into a problem when using this test:

```
go build -v
mkdir mm && mv croc mm
cd mm
touch .gitignore
echo "croc" > .gitignore
touch test.txt
echo "test" > test.txt
./croc send *
```

this time its a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x69506f]

goroutine 1 [running]:
github.com/sabhiram/go-gitignore.(*GitIgnore).MatchesPathHow(0x0, {0x7ffe990e702f?, 0x7fcbe8?})
	/home/zns/go/pkg/mod/github.com/sabhiram/go-gitignore@v0.0.0-20210923224102-525f6e181f06/ignore.go:223 +0x6f
github.com/sabhiram/go-gitignore.(*GitIgnore).MatchesPath(...)
	/home/zns/go/pkg/mod/github.com/sabhiram/go-gitignore@v0.0.0-20210923224102-525f6e181f06/ignore.go:210
github.com/schollz/croc/v9/src/croc.GetFilesInfo({0xc000486320?, 0x2, 0x4?}, 0x0)
	/home/zns/Documents/croc/src/croc/croc.go:323 +0x468
github.com/schollz/croc/v9/src/cli.send(0xc0000ad800)
	/home/zns/Documents/croc/src/cli/cli.go:282 +0xc4b
github.com/schollz/cli/v2.(*Command).Run(0xc000134480, 0xc0000ad500)
	/home/zns/go/pkg/mod/github.com/schollz/cli/v2@v2.2.1/command.go:163 +0x5c8
github.com/schollz/cli/v2.(*App).RunContext(0xc00016f200, {0x801ca8?, 0xc0000b2020}, {0xc0000ac000, 0x4, 0x4})
	/home/zns/go/pkg/mod/github.com/schollz/cli/v2@v2.2.1/app.go:306 +0xb36
github.com/schollz/cli/v2.(*App).Run(...)
	/home/zns/go/pkg/mod/github.com/schollz/cli/v2@v2.2.1/app.go:217
github.com/schollz/croc/v9/src/cli.Run()
	/home/zns/Documents/croc/src/cli/cli.go:140 +0x13dc
main.main()
	/home/zns/Documents/croc/main.go:30 +0x19
```

**Comment by PThorpe92** [comment]
thats wild I ran that so many times... My fault, marking as draft

**Pull request closed without merging**